thamos/lib.py

```diff
@@ -97,6 +96,27 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def get_advise_status(analysis_id: str) -> typing.Optional[dict]:
+    """Handle the the multiple response types available while asking for result of a advise."""
+    config = Configuration()
+    host = thoth_config.explicit_host
+    if not host:
+        thoth_config.load_config()
+        host = thoth_config.content.get("host") or config.host
+    response = requests.Session().get(
+        f"https://{host}/api/v1/advise/python/{analysis_id}"
+    )
+    if response.status_code == 202:
+        res = json.loads(response.text)
+        return res
+    # If results are already available we set status to 1.
+    elif response.status_code == 200:
+        res["status"] = {"finished_at": 1}
```

Comment:
I'm wondering if we could handle this in a prettier way. What about introducing something like `advise_is_ready` that would return `True`/`False`?

Reply:
@fridex Seems it's being called at a couple of different points like [link](https://github.com/thoth-station/thamos/blob/1a6650ef3db3a387a9bafb0b2921471d033ecf79/thamos/lib.py#L488) and [link](https://github.com/thoth-station/thamos/blob/1a6650ef3db3a387a9bafb0b2921471d033ecf79/thamos/lib.py#L568)

Reply:
Actually I will change all these to call that `analysis_is_ready` method, so that internally they call the new `get_document` method in user-api, no more calling the status endpoint.